sdk/src/main/scala/com/horizen/validation/crosschain/sender/AbstractCrossChainMessageValidator.scala

```diff
@@ -0,0 +1,78 @@
+package com.horizen.validation.crosschain.sender
+
+import com.horizen.block.SidechainBlock
+import com.horizen.box.CrossChainMessageBox
+import com.horizen.cryptolibprovider.CryptoLibProvider
+import com.horizen.params.NetworkParams
+import com.horizen.sc2sc.{CrossChainMessageHash, Sc2ScConfigurator}
+import com.horizen.storage.SidechainStateStorage
+import com.horizen.transaction.MC2SCAggregatedTransaction
+import com.horizen.validation.crosschain.CrossChainValidator
+import com.horizen.{SidechainState, SidechainTypes}
+
+import scala.collection.JavaConverters._
+
+class AbstractCrossChainMessageValidator(
```

Comment:
This class is extended by the custom message validator from the application side. As we discussed we can add a **protected applicationValidate** that is called inside the validate and it is implemented by the custom validator.